File: ccms/src/svc/report/cms_set_brcb_result.c
#include "svc/report/cms_set_brcb_result.h"
#include "data/scalar/cms_boolean.h"
#include "data/common/cms_service_error.h"

int cms_set_brcb_result_encode_stream(per_stream_t *s, const cms_set_brcb_result_t *v) {
    if (!v) return CMS_ERR;
    int err;

    /* 1. error — ServiceError OPTIONAL */
    { int p = (v->error_present && v->error_present->value) && v->error; cms_boolean_t b={.value=p}; err=cms_boolean_encode_stream(s,&b); if(err)return err; if(p){err=cms_service_error_encode_stream(s,v->error);if(err)return err;} }

    /* 2. rptIdErr — ServiceError OPTIONAL */
    { int p = (v->rpt_id_err_present && v->rpt_id_err_present->value) && v->rpt_id_err; cms_boolean_t b={.value=p}; err=cms_boolean_encode_stream(s,&b); if(err)return err; if(p){err=cms_service_error_encode_stream(s,v->rpt_id_err);if(err)return err;} }

    /* 3. rptEnaErr — ServiceError OPTIONAL */
    { int p = (v->rpt_ena_err_present && v->rpt_ena_err_present->value) && v->rpt_ena_err; cms_boolean_t b={.value=p}; err=cms_boolean_encode_stream(s,&b); if(err)return err; if(p){err=cms_service_error_encode_stream(s,v->rpt_ena_err);if(err)return err;} }

    /* 4. datSetErr — ServiceError OPTIONAL */
    { int p = (v->dat_set_err_present && v->dat_set_err_present->value) && v->dat_set_err; cms_boolean_t b={.value=p}; err=cms_boolean_encode_stream(s,&b); if(err)return err; if(p){err=cms_service_error_encode_stream(s,v->dat_set_err);if(err)return err;} }

    /* 5. optFldsErr — ServiceError OPTIONAL */
    { int p = (v->opt_flds_err_present && v->opt_flds_err_present->value) && v->opt_flds_err; cms_boolean_t b={.value=p}; err=cms_boolean_encode_stream(s,&b); if(err)return err; if(p){err=cms_service_error_encode_stream(s,v->opt_flds_err);if(err)return err;} }

    /* 6. bufTmErr — ServiceError OPTIONAL */
    { int p = (v->buf_tm_err_present && v->buf_tm_err_present->value) && v->buf_tm_err; cms_boolean_t b={.value=p}; err=cms_boolean_encode_stream(s,&b); if(err)return err; if(p){err=cms_service_error_encode_stream(s,v->buf_tm_err);if(err)return err;} }

    /* 7. trgOpsErr — ServiceError OPTIONAL */
    { int p = (v->trg_ops_err_present && v->trg_ops_err_present->value) && v->trg_ops_err; cms_boolean_t b={.value=p}; err=cms_boolean_encode_stream(s,&b); if(err)return err; if(p){err=cms_service_error_encode_stream(s,v->trg_ops_err);if(err)return err;} }

    /* 8. intgPdErr — ServiceError OPTIONAL */
    { int p = (v->intg_pd_err_present && v->intg_pd_err_present->value) && v->intg_pd_err; cms_boolean_t b={.value=p}; err=cms_boolean_encode_stream(s,&b); if(err)return err; if(p){err=cms_service_error_encode_stream(s,v->intg_pd_err);if(err)return err;} }

    /* 9. giErr — ServiceError OPTIONAL */
    { int p = (v->gi_err_present && v->gi_err_present->value) && v->gi_err; cms_boolean_t b={.value=p}; err=cms_boolean_encode_stream(s,&b); if(err)return err; if(p){err=cms_service_error_encode_stream(s,v->gi_err);if(err)return err;} }

    /* 10. purgeBufErr — ServiceError OPTIONAL */
    { int p = (v->purge_buf_err_present && v->purge_buf_err_present->value) && v->purge_buf_err; cms_boolean_t b={.value=p}; err=cms_boolean_encode_stream(s,&b); if(err)return err; if(p){err=cms_service_error_encode_stream(s,v->purge_buf_err);if(err)return err;} }

    /* 11. entryIdErr — ServiceError OPTIONAL */
    { int p = (v->entry_id_err_present && v->entry_id_err_present->value) && v->entry_id_err; cms_boolean_t b={.value=p}; err=cms_boolean_encode_stream(s,&b); if(err)return err; if(p){err=cms_service_error_encode_stream(s,v->entry_id_err);if(err)return err;} }

    /* 12. resvTmsErr — ServiceError OPTIONAL */
    { int p = (v->resv_tms_err_present && v->resv_tms_err_present->value) && v->resv_tms_err; cms_boolean_t b={.value=p}; err=cms_boolean_encode_stream(s,&b); if(err)return err; if(p){err=cms_service_error_encode_stream(s,v->resv_tms_err);if(err)return err;} }

    return CMS_OK;
}

int cms_set_brcb_result_decode_stream(per_stream_t *s, cms_set_brcb_result_t *v) {
    if (!v) return CMS_ERR;
    int err;

    /* 1. error OPTIONAL */
    { cms_boolean_t b={0}; err=cms_boolean_decode_stream(s,&b); if(err)return err; if(v->error_present)v->error_present->value=b.value; if(b.value&&v->error){err=cms_service_error_decode_stream(s,v->error);if(err)return err;} }

    /* 2. rptIdErr OPTIONAL */
    { cms_boolean_t b={0}; err=cms_boolean_decode_stream(s,&b); if(err)return err; if(v->rpt_id_err_present)v->rpt_id_err_present->value=b.value; if(b.value&&v->rpt_id_err){err=cms_service_error_decode_stream(s,v->rpt_id_err);if(err)return err;} }

    /* 3. rptEnaErr OPTIONAL */
    { cms_boolean_t b={0}; err=cms_boolean_decode_stream(s,&b); if(err)return err; if(v->rpt_ena_err_present)v->rpt_ena_err_present->value=b.value; if(b.value&&v->rpt_ena_err){err=cms_service_error_decode_stream(s,v->rpt_ena_err);if(err)return err;} }

    /* 4. datSetErr OPTIONAL */
    { cms_boolean_t b={0}; err=cms_boolean_decode_stream(s,&b); if(err)return err; if(v->dat_set_err_present)v->dat_set_err_present->value=b.value; if(b.value&&v->dat_set_err){err=cms_service_error_decode_stream(s,v->dat_set_err);if(err)return err;} }

    /* 5. optFldsErr OPTIONAL */
    { cms_boolean_t b={0}; err=cms_boolean_decode_stream(s,&b); if(err)return err; if(v->opt_flds_err_present)v->opt_flds_err_present->value=b.value; if(b.value&&v->opt_flds_err){err=cms_service_error_decode_stream(s,v->opt_flds_err);if(err)return err;} }

    /* 6. bufTmErr OPTIONAL */
    { cms_boolean_t b={0}; err=cms_boolean_decode_stream(s,&b); if(err)return err; if(v->buf_tm_err_present)v->buf_tm_err_present->value=b.value; if(b.value&&v->buf_tm_err){err=cms_service_error_decode_stream(s,v->buf_tm_err);if(err)return err;} }

    /* 7. trgOpsErr OPTIONAL */
    { cms_boolean_t b={0}; err=cms_boolean_decode_stream(s,&b); if(err)return err; if(v->trg_ops_err_present)v->trg_ops_err_present->value=b.value; if(b.value&&v->trg_ops_err){err=cms_service_error_decode_stream(s,v->trg_ops_err);if(err)return err;} }

    /* 8. intgPdErr OPTIONAL */
    { cms_boolean_t b={0}; err=cms_boolean_decode_stream(s,&b); if(err)return err; if(v->intg_pd_err_present)v->intg_pd_err_present->value=b.value; if(b.value&&v->intg_pd_err){err=cms_service_error_decode_stream(s,v->intg_pd_err);if(err)return err;} }

    /* 9. giErr OPTIONAL */
    { cms_boolean_t b={0}; err=cms_boolean_decode_stream(s,&b); if(err)return err; if(v->gi_err_present)v->gi_err_present->value=b.value; if(b.value&&v->gi_err){err=cms_service_error_decode_stream(s,v->gi_err);if(err)return err;} }

    /* 10. purgeBufErr OPTIONAL */
    { cms_boolean_t b={0}; err=cms_boolean_decode_stream(s,&b); if(err)return err; if(v->purge_buf_err_present)v->purge_buf_err_present->value=b.value; if(b.value&&v->purge_buf_err){err=cms_service_error_decode_stream(s,v->purge_buf_err);if(err)return err;} }

    /* 11. entryIdErr OPTIONAL */
    { cms_boolean_t b={0}; err=cms_boolean_decode_stream(s,&b); if(err)return err; if(v->entry_id_err_present)v->entry_id_err_present->value=b.value; if(b.value&&v->entry_id_err){err=cms_service_error_decode_stream(s,v->entry_id_err);if(err)return err;} }

    /* 12. resvTmsErr OPTIONAL */
    { cms_boolean_t b={0}; err=cms_boolean_decode_stream(s,&b); if(err)return err; if(v->resv_tms_err_present)v->resv_tms_err_present->value=b.value; if(b.value&&v->resv_tms_err){err=cms_service_error_decode_stream(s,v->resv_tms_err);if(err)return err;} }

    return CMS_OK;
}

File: jcms/jcms-core/src/main/java/com/ysh/jcms/svc/report/CmsSetBrcbResult.java
package com.ysh.jcms.svc.report;

import com.ysh.jcms.core.CmsTypeOld;
import com.ysh.jcms.data.common.CmsServiceError;
import com.ysh.jcms.data.scalar.CmsBoolean;
import java.util.Arrays;
import java.util.List;

/**
 * SetBRCBResult ::= SEQUENCE { error [0] IMPLICIT ServiceError OPTIONAL, rptID
 * [1] IMPLICIT ServiceError OPTIONAL, rptEna [2] IMPLICIT ServiceError
 * OPTIONAL, datSet [3] IMPLICIT ServiceError OPTIONAL, optFlds [5] IMPLICIT
 * ServiceError OPTIONAL, bufTm [6] IMPLICIT ServiceError OPTIONAL, trgOps [8]
 * IMPLICIT ServiceError OPTIONAL, intgPd [9] IMPLICIT ServiceError OPTIONAL, gi
 * [10] IMPLICIT ServiceError OPTIONAL, purgeBuf [11] IMPLICIT ServiceError
 * OPTIONAL, entryID [12] IMPLICIT ServiceError OPTIONAL, resvTms [14] IMPLICIT
 * ServiceError OPTIONAL } — 8.7.3
 */
public class CmsSetBrcbResult extends CmsTypeOld {

    public CmsBoolean errorPresent;
    public CmsServiceError error; /* OPTIONAL */
    public CmsBoolean rptIdErrPresent;
    public CmsServiceError rptIdErr; /* OPTIONAL */
    public CmsBoolean rptEnaErrPresent;
    public CmsServiceError rptEnaErr; /* OPTIONAL */
    public CmsBoolean datSetErrPresent;
    public CmsServiceError datSetErr; /* OPTIONAL */
    public CmsBoolean optFldsErrPresent;
    public CmsServiceError optFldsErr; /* OPTIONAL */
    public CmsBoolean bufTmErrPresent;
    public CmsServiceError bufTmErr; /* OPTIONAL */
    public CmsBoolean trgOpsErrPresent;
    public CmsServiceError trgOpsErr; /* OPTIONAL */
    public CmsBoolean intgPdErrPresent;
    public CmsServiceError intgPdErr; /* OPTIONAL */
    public CmsBoolean giErrPresent;
    public CmsServiceError giErr; /* OPTIONAL */
    public CmsBoolean purgeBufErrPresent;
    public CmsServiceError purgeBufErr; /* OPTIONAL */
    public CmsBoolean entryIdErrPresent;
    public CmsServiceError entryIdErr; /* OPTIONAL */
    public CmsBoolean resvTmsErrPresent;
    public CmsServiceError resvTmsErr; /* OPTIONAL */

    public CmsSetBrcbResult() {
        this.errorPresent = new CmsBoolean();
        this.error = new CmsServiceError();
        this.rptIdErrPresent = new CmsBoolean();
        this.rptIdErr = new CmsServiceError();
        this.rptEnaErrPresent = new CmsBoolean();
        this.rptEnaErr = new CmsServiceError();
        this.datSetErrPresent = new CmsBoolean();
        this.datSetErr = new CmsServiceError();
        this.optFldsErrPresent = new CmsBoolean();
        this.optFldsErr = new CmsServiceError();
        this.bufTmErrPresent = new CmsBoolean();
        this.bufTmErr = new CmsServiceError();
        this.trgOpsErrPresent = new CmsBoolean();
        this.trgOpsErr = new CmsServiceError();
        this.intgPdErrPresent = new CmsBoolean();
        this.intgPdErr = new CmsServiceError();
        this.giErrPresent = new CmsBoolean();
        this.giErr = new CmsServiceError();
        this.purgeBufErrPresent = new CmsBoolean();
        this.purgeBufErr = new CmsServiceError();
        this.entryIdErrPresent = new CmsBoolean();
        this.entryIdErr = new CmsServiceError();
        this.resvTmsErrPresent = new CmsBoolean();
        this.resvTmsErr = new CmsServiceError();
    }

    public CmsSetBrcbResult errorPresent(boolean v) {
        this.errorPresent.value(v);
        return this;
    }
    public CmsSetBrcbResult error(int v) {
        this.error.value(v);
        return this;
    }
    public CmsSetBrcbResult rptIdErrPresent(boolean v) {
        this.rptIdErrPresent.value(v);
        return this;
    }
    public CmsSetBrcbResult rptIdErr(int v) {
        this.rptIdErr.value(v);
        return this;
    }
    public CmsSetBrcbResult rptEnaErrPresent(boolean v) {
        this.rptEnaErrPresent.value(v);
        return this;
    }
    public CmsSetBrcbResult rptEnaErr(int v) {
        this.rptEnaErr.value(v);
        return this;
    }
    public CmsSetBrcbResult datSetErrPresent(boolean v) {
        this.datSetErrPresent.value(v);
        return this;
    }
    public CmsSetBrcbResult datSetErr(int v) {
        this.datSetErr.value(v);
        return this;
    }
    public CmsSetBrcbResult optFldsErrPresent(boolean v) {
        this.optFldsErrPresent.value(v);
        return this;
    }
    public CmsSetBrcbResult optFldsErr(int v) {
        this.optFldsErr.value(v);
        return this;
    }
    public CmsSetBrcbResult bufTmErrPresent(boolean v) {
        this.bufTmErrPresent.value(v);
        return this;
    }
    public CmsSetBrcbResult bufTmErr(int v) {
        this.bufTmErr.value(v);
        return this;
    }
    public CmsSetBrcbResult trgOpsErrPresent(boolean v) {
        this.trgOpsErrPresent.value(v);
        return this;
    }
    public CmsSetBrcbResult trgOpsErr(int v) {
        this.trgOpsErr.value(v);
        return this;
    }
    public CmsSetBrcbResult intgPdErrPresent(boolean v) {
        this.intgPdErrPresent.value(v);
        return this;
    }
    public CmsSetBrcbResult intgPdErr(int v) {
        this.intgPdErr.value(v);
        return this;
    }
    public CmsSetBrcbResult giErrPresent(boolean v) {
        this.giErrPresent.value(v);
        return this;
    }
    public CmsSetBrcbResult giErr(int v) {
        this.giErr.value(v);
        return this;
    }
    public CmsSetBrcbResult purgeBufErrPresent(boolean v) {
        this.purgeBufErrPresent.value(v);
        return this;
    }
    public CmsSetBrcbResult purgeBufErr(int v) {
        this.purgeBufErr.value(v);
        return this;
    }
    public CmsSetBrcbResult entryIdErrPresent(boolean v) {
        this.entryIdErrPresent.value(v);
        return this;
    }
    public CmsSetBrcbResult entryIdErr(int v) {
        this.entryIdErr.value(v);
        return this;
    }
    public CmsSetBrcbResult resvTmsErrPresent(boolean v) {
        this.resvTmsErrPresent.value(v);
        return this;
    }
    public CmsSetBrcbResult resvTmsErr(int v) {
        this.resvTmsErr.value(v);
        return this;
    }

    @Override
    public List<? extends CmsTypeOld> children() {
        return Arrays.asList(errorPresent, error, rptIdErrPresent, rptIdErr, rptEnaErrPresent, rptEnaErr, datSetErrPresent, datSetErr,
                optFldsErrPresent, optFldsErr, bufTmErrPresent, bufTmErr, trgOpsErrPresent, trgOpsErr, intgPdErrPresent, intgPdErr,
                giErrPresent, giErr, purgeBufErrPresent, purgeBufErr, entryIdErrPresent, entryIdErr, resvTmsErrPresent, resvTmsErr);
    }
}
